File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>mp;
        vector<int>ans;        
        for(int i=0;i<nums.size();i++)
        {
            if(mp.count(target-nums[i])==1)
                return {mp[target-nums[i]] , i};
            
            else mp[nums[i]]=i; 
        }
        return ans;
        
    }
};

File: 2025-and-before/1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let mp = new Map();
    
    for(let i=0;i<nums.length;i++)
    {
        if(target - nums[i] in mp)
            return [mp[target - nums[i]], i];
        
        mp[nums[i]]=i;
        
    }
    
};